pypsa/optimization/optimize.py

```diff
@@ -22,16 +22,21 @@
 )
 from pypsa.optimization.common import get_strongly_meshed_buses, set_from_frame
 from pypsa.optimization.constraints import (
```

Comment:
I see a lot of new functions; we should reduce them and avoid changing the name to the existing ones

Reply:
In my opinion we cannot. As you can see, we now have 5 functions, and:
1) Before PyPSA used 3 functions, which I've simply renamed, since the previous names were based on the idea that committability and extendability are not compatible and now it could lead to misunderstandings.
2) One of the 3 functions is used twice with different names to avoid misunderstandings when choosing the right idx (see attached file).
3) I have just added a single new feature that could potentially replace 2 out of 4 of the previous features, but we prefer not to do so in order to maintain backwards compatibility.

To explain the need for these 5 functions I have prepared the attached pdf file. I have also included the Google Drive link so that you can comment on the file if you prefer: https://docs.google.com/document/d/1CUDymdWgfhw56-dcihNnF0sKwZSMe1W9/edit?usp=sharing&ouid=112809165806691547710&rtpof=true&sd=true

[PR_comment.docx.pdf](https://github.com/user-attachments/files/16735809/PR_comment.docx.pdf)